crates/dids/src/resolver.rs

```diff
@@ -0,0 +1,109 @@
+use crate::method::jwk::DidJwk;
+use crate::method::DidMethod;
+use serde::{Deserialize, Serialize};
+use ssi_dids::did_resolve::{
+    DocumentMetadata as DidDocumentMetadata, ResolutionMetadata as DidResolutionMetadata,
+};
+use ssi_dids::Document as DidDocument;
+
+pub struct DidResolver;
+
+impl DidResolver {
+    /// Resolves a DID URI, using the appropriate DID method, to a DID Document.
+    pub async fn resolve_uri(did_uri: &str) -> DidResolutionResult {
+        let method_name = match DidResolver::method_name(did_uri) {
+            Some(method_name) => method_name,
+            None => return DidResolutionResult::from_error(ERROR_INVALID_DID),
+        };
+
+        match method_name {
+            DidJwk::NAME => DidJwk::resolve_uri(did_uri).await,
+            _ => return DidResolutionResult::from_error(ERROR_METHOD_NOT_SUPPORTED),
+        }
+    }
+
+    /// Returns the method name of a DID URI, if the provided DID URI is valid, `None` otherwise.
+    fn method_name(did_uri: &str) -> Option<&str> {
+        let parts: Vec<&str> = did_uri.split(':').collect();
+        if parts.len() < 3 || parts[0] != "did" {
+            return None;
+        };
+
+        Some(parts[1])
+    }
```

Comment:
I would expect this code to be inside the `Did` trait. Thoughts?

Reply:
The `Did` trait defines behavior for `Did` implementations. Each implementation is already defining the name it's implementing. This function is computing the method name from some arbitrarily provided string.

We could definitely move this into a `utils` module or something similar.

Reply:
The answer is never "utils"